AppDevAnnouncements/Classes/NotificationViewController.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  NotificationViewController.swift
+//  AppDevAnnouncements
+//
+//  Created by Gonzalo Gonzalez on 3/11/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+
+protocol NotificationDelegate: class {
+    func present(_ notificationViewController: NotificationViewController)
+}
+
+class NotificationViewController: UIViewController {
+
+    var notificationView: NotificationView!
+
+    var visual : String?
+    var subject : String?
+    var body : String?
+    var ctaText : String?
+    var ctaAction : String?
+
+    var presenter: UIViewController?
+    weak var delegate : NotificationDelegate?
+
+    convenience init(visual: String?, subject: String?, body: String?, ctaText: String?, ctaAction: String?, delegate: NotificationDelegate) {
+        self.init(nibName:nil, bundle:nil)
+        self.visual = visual
+        self.subject = subject
+        self.body = body
+        self.ctaText = ctaText
+        self.ctaAction = ctaAction
+        self.delegate = delegate
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .clear
+
+        notificationView = NotificationView()
+        if let unwrappedVisual = visual {
+            notificationView.visualImageView.image = UIImage(named: unwrappedVisual)
+        }
+        if let unwrappedSubject = subject {
+            notificationView.subjectLabel.text = unwrappedSubject
+        }
+        if let unwrappedBody = body {
+            notificationView.bodyTextView.text = unwrappedBody
+        }
+        if let unwrappedCTAText = ctaText {
```

Comment:
I don't know how I feel about having if lets for every field in this notification haha.

Reply:
+1. I feel like if they really r all gonna be optional, then it should be an all or nothing kinda thing

Reply:
Yeah I agree, the if let after if let looked disgusting. All or nothing.